lib/inventory_refresh/application_record_iterator.rb

```diff
@@ -26,11 +26,9 @@ def initialize(inventory_collection: nil, manager_uuids_set: nil, iterator: nil,
     #
     # @param batch_size [Integer] A batch size we want to fetch from DB
     # @yield Code processing the batches
-    def find_in_batches(batch_size: 1000)
+    def find_in_batches(batch_size: 1000, &block)
       if iterator
-        iterator.call do |batch|
-          yield(batch)
-        end
+        iterator.call(&block)
       elsif query
         manager_uuids_set.each_slice(batch_size) do |batch|
           yield(query.where(inventory_collection.targeted_selection_for(batch)))
```

Comment:
Strange that it didn't use block.call here.

Reply:
also, `yield` is faster than `call`. So using a yield would probably be preferred